tests/scripts/remote_monitoring_tests/rest_apis/test_update_recommendations.py

```diff
@@ -0,0 +1,273 @@
+
+        response = update_recommendations(experiment_name, None, end_time)
+        data = response.json()
+        assert response.status_code == SUCCESS_STATUS_CODE
+
+        # Invoke list recommendations for the specified experiment
+        response = list_recommendations(experiment_name)
+        assert response.status_code == SUCCESS_200_STATUS_CODE
+        list_reco_json = response.json()
+
+        # Validate the json against the json schema
+        errorMsg = validate_list_reco_json(list_reco_json, list_reco_json_schema)
+        assert errorMsg == ""
+
+        # Validate the json values
+        create_exp_json = read_json_data_from_file(create_exp_json_file)
+        update_results_json = []
+        update_results_json.append(result_json_arr[len(result_json_arr) - 1])
+
+        expected_duration_in_hours = SHORT_TERM_DURATION_IN_HRS_MAX
+        validate_reco_json(create_exp_json[0], update_results_json, list_reco_json[0], expected_duration_in_hours)
+
+    # Delete all the experiments
+    for i in range(num_exps):
+        json_file = "/tmp/create_exp_" + str(i) + ".json"
+        response = delete_experiment(json_file)
+        print("delete exp = ", response.status_code)
+        assert response.status_code == SUCCESS_STATUS_CODE
+
+
+@pytest.mark.negative
+def test_update_recommendations_without_experiment_name_end_time(cluster_type):
+    form_kruize_url(cluster_type)
+    response = update_recommendations(None, None, None)
+    data = response.json()
+    assert response.status_code == ERROR_STATUS_CODE
+    assert data['message'] == UPDATE_RECOMMENDATIONS_MANDATORY_DEFAULT_MESSAGE
+
+
+@pytest.mark.negative
+def test_update_recommendations_without_end_time(cluster_type):
+    form_kruize_url(cluster_type)
+    experiment_name = "test123"
+    response = update_recommendations(experiment_name, None, None)
+    data = response.json()
+    assert response.status_code == ERROR_STATUS_CODE
+    assert data['message'] == UPDATE_RECOMMENDATIONS_MANDATORY_INTERVAL_END_DATE
+
+
+@pytest.mark.negative
+def test_update_recommendations_with_unknown_experiment_name_and_end_time(cluster_type):
+    form_kruize_url(cluster_type)
+    experiment_name = "test123"
+    end_time = "2023-01-02T00:15:00.000Z"
+    response = update_recommendations(experiment_name, None, end_time)
+    data = response.json()
+    assert response.status_code == ERROR_STATUS_CODE
+    assert data['message'] == UPDATE_RECOMMENDATIONS_DATA_NOT_FOUND
+
+
+@pytest.mark.negative
+def test_update_recommendations_with_end_time_precede_start_time(cluster_type):
+    form_kruize_url(cluster_type)
+    experiment_name = "test123"
+    start_time = "2023-01-03T00:15:00.000Z"
+    end_time = "2023-01-02T00:15:00.000Z"
+    response = update_recommendations(experiment_name, start_time, end_time)
+    data = response.json()
+    assert response.status_code == ERROR_STATUS_CODE
+    assert data['message'] == UPDATE_RECOMMENDATIONS_START_TIME_PRECEDE_END_TIME
```

Comment:
Please update the API doc with this error message

Reply:
Can you please share the results for all these tests.

Reply:
Done

Reply:
> Can you please share the results for all these tests.

https://drive.google.com/drive/folders/17e1KE3eZKi8Waf3o7uvlMQ1SqJC8cpfW?usp=sharing